script/infra-setup

```diff
@@ -1,53 +1,141 @@
 
-wget https://raw.githubusercontent.com/atulabhi/litmus/v0.7-RC1/providers/openebs/installers/operator/master/litmusbook/openebs_setup.yaml
+## TODO: Add logic to replace job ENV based on run instance intent 
+## TODO: Add logic to add ENV for run_instance_metadata/RunID
 
-kubectl apply -f openebs_setup.yaml
+#################
+## RUNNER MAIN ##
+#################
 
-sleep 120
+echo "Running litmus test for openebs deploy.."
 
-kubectl get po --all-namespaces
+jobNameCmd="kubectl get jobs -n litmus --no-headers -o jsonpath='{.items[?(@.metadata.labels.provider==\"openebs-setup\")].metadata.name}'"
 
+job_deploy_out=$(kubectl create -f providers/openebs/installers/operator/master/litmusbook/openebs_setup.yaml) && \
+job_name=$(eval $jobNameCmd); retcode=$?
+error_handler $retcode msg="Unable to run litmusbook, exiting" action="exit"
 
-litmus_pod=$(kubectl get pod -a -n litmus --no-headers -o custom-columns=:metadata.name)
-status_cmd='kubectl logs $litmus_pod -n litmus -c ansibletest'
-while [[ $(kubectl get pod $litmus_pod --no-headers -n litmus -o custom-columns=:status.phase) =~ 'Running' ]]
-do
-   echo "Still Running sir..."
+## Obtain the litmus pod name 
+
+litmusPodCmd="kubectl get pod --no-headers -n litmus -o jsonpath='{.items[?(@.metadata.labels.job-name==\"$job_name\")].metadata.name}'"
+litmus_pod=$(eval $litmusPodCmd); retcode=$?
+error_handler $retcode msg="Unable to find litmus test runner pod, exiting" action="exit"
+
+## Wait till the ansibletest container terminates && also confirm job completion status. This is done to ensure
+## that execution of auxiliary containers such as loggers is completed. Getting the ansibletest ccontainer to completed state 
+## satisfies the "necessary" condition for test job completion
+
+containerStateCmd="kubectl get pod ${litmus_pod} -n litmus -o jsonpath='{.status.containerStatuses[?(@.name==\"ansibletest\")].state}'"
+jobStateCmd="kubectl get pod ${litmus_pod} --no-headers -n litmus -o custom-columns=:status.phase"
+
+## TODO: Consider cases where litmus pod is evicted
+
+while [[ ! $(eval $containerStateCmd) =~ 'terminated' ]]; do
+  sleep 1
+done
+
+while [[ $(eval $jobStateCmd) =~ 'Running' ]]; do
    sleep 1
 done
-if [[ ! $(eval $status_cmd) =~ 'FAILED!' ]]; then
-   TestStatus='Pass'
+
+echo "Litmus test run Job has completed"
+task_delimiter;
+
+## Printing the test logs & cluster state for debug purposes 
+
+echo "Dumping Litmus test pod logs for debug"; echo ""
+kubectl logs $litmus_pod -n litmus -c ansibletest
+task_delimiter;
+
+echo "Dumping state of cluster post job run"; echo ""
+kubectl get pods --all-namespaces
+kubectl get pvc --all-namespaces
+kubectl get sc 
+
+task_delimiter;
+
+
+## Check the test status & result from the litmus result custom resource
+
+testStatus=$(kubectl get lr $test_name --no-headers -o custom-columns=:spec.testStatus.phase) && \
+testResult=$(kubectl get lr $test_name --no-headers -o custom-columns=:spec.testStatus.result); retcode=$?
```

Comment:
@ksachit, I think the test name variable for the openebs setup litmus book is $installation_test_name. Can we change the variable?